src/rdkafka_cgrp.c

```diff
@@ -2941,14 +2953,25 @@ rd_kafka_cgrp_consumer_assignment_with_metadata(
                                 topic_name = rd_strdup(rktpar->topic);
                 }
 
-                if (likely(topic_name != NULL)) {
+                // If topic name is found and partition exists in metadata
+                if (likely(topic_name != NULL) && partition_found) {
                         rd_kafka_topic_partition_list_add_with_topic_name_and_id(
                             assignment_with_metadata, request_topic_id,
                             topic_name, assignment->elems[i].partition);
                         rd_free(topic_name);
                         continue;
                 }
 
+                if (!partition_found)
+                        rd_kafka_dbg(
+                            rkcg->rkcg_rk, CGRP, "HEARTBEAT",
+                            "Partition assigned to this consumer is not "
+                            "present in cached metadata for topic id: %s. "
+                            "This may indicate that the topic's partition "
+                            "count has increased and metadata needs to be "
+                            "refreshed. ",
+                            rd_kafka_Uuid_base64str(&request_topic_id));
```

Comment:
Debug statement should not be so verbose. I think its sufficient to add something like "Found new partition for the topic %s. Updating metadata.". Add topic name as well instead of just topic id.